exercises/queen-attack/src/example/java/QueenAttackCalculator.java

```diff
@@ -0,0 +1,86 @@
+import java.util.Arrays;
+
+public final class QueenAttackCalculator {
+
+    private final int[] queen1Coordinates;
+
+    private final int[] queen2Coordinates;
+
+    public QueenAttackCalculator(final int[] queen1Coordinates, final int[] queen2Coordinates)
```

Comment:
subject "SAN":
There is *not* currently a mechanism for extra-credit, per se.  It's merely a mention of where the exercising programmer could go from here.

Personally, I'd like to see this exercise be defined as using SAN, but that takes rough consensus across the tracks (read: not happening, today).

It could be an interesting exercise to declare two contracts (one with cartesian coordinate and another for SAN coordinates) and a test class for both of those interfaces.  The SAN test class would have to make it clear that it is optional; there ought to be additional readme (`EXTRA-CREDIT.md`?) to provide a proper framing.

None of that is at all required for this PR.  I'm just thinking out loud, now.